akiraux/Akira
# Basic Paths Feature

## Summary / How this PR fixes the problem?
- Created action to insert a path when user presses ```p``` or from header bar.
- Created a **PathEditMode** to for creating a new path. In future PRs, this can be reused to edit paths of other CanvasItems when user double clicks on them.
- Created a **ViewLayerPath** so that users can easily visualise the points they are inserting in the path

## Steps to Test

- Press ```p``` or click on **Vector** in the insert menu
- Click randomly on the canvas. The points where you click will be highlighted in blue. This is the work of ViewLayerPath. All these points will be connected using lines.
- Press ```Esc``` to exit **PathEditMode**. The blue dots from **ViewLayerPath** should now be replaced by nobs and selection box.

## Screenshots 
![animation](https://user-images.githubusercontent.com/64722310/131217522-f3c1449e-f9c1-49bb-9eff-ac4b9c0c2323.gif)


## Known Issues / Things To Do

The following features will be implemented step by step in small PRs to make developing and reviewing easier.

- Add curves using the ```Ctrl```-```Alt``` method.
- Double clicking on an item should enter user in PathEditMode
- Scaling paths doesn't work either.
